buildconfig/CMake/WindowsNSIS.cmake

```diff
@@ -176,12 +194,49 @@ install ( FILES ${QT_PLUGIN_DIR}/sqldrivers/qsqlite4.dll DESTINATION plugins/qt4
+install ( FILES ${PROJECT_BINARY_DIR}/mantidpython.bat.install DESTINATION bin RENAME mantidpython.bat )
+
+# On Windows we don't use the setuptools install executable at the moment, because it is
+# generated with a hard coded path to the build Python, that fails to run on
+# other machines, or if the build Python is moved. Instead we have a Python script to
+# run the workbench ourselves and a Batch and PowerShell wrappers to run it with the
+# correct Python from Mantid's installation directory
+install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/buildconfig/CMake/Packaging/launch_workbench.bat DESTINATION bin )
+install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/buildconfig/CMake/Packaging/launch_workbench.pyw DESTINATION bin )
+
+if ( ENABLE_WORKBENCH AND PACKAGE_WORKBENCH )
+  find_program(POWERSHELL_AVAILABLE NAMES "powershell")
+  # Name of the workbench executable without any extensions
+  set(_workbench_base_name launch_workbench)
+  # Installs the PowerShell run script in the bin directory
+  install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/buildconfig/CMake/Packaging/${_workbench_base_name}.ps1 DESTINATION bin )
+
+  if ( POWERSHELL_AVAILABLE )
+    message(STATUS "Found PowerShell: ${POWERSHELL_AVAILABLE}")
+    # Add the extensions for the executable after it is generated in the build directory
+    set(_workbench_executable_install_name ${_workbench_base_name}.exe.install)
+    # Generate an executable from the PowerShell script
+    execute_process(COMMAND powershell.exe -version 2.0 -noprofile -windowstyle hidden ${THIRD_PARTY_DIR}/bin/ps2exe.ps1 -inputFile ${CMAKE_CURRENT_SOURCE_DIR}/buildconfig/CMake/Packaging/${_workbench_base_name}.ps1 -outputFile ${CMAKE_CURRENT_BINARY_DIR}/${_workbench_executable_install_name} -x64 -runtime2 -noconsole RESULT_VARIABLE _workbench_powershell_return_code OUTPUT_VARIABLE _workbench_powershell_output)
 
-###########################################################################	
-# Icons for shortcuts	
-###########################################################################	
+    # If the EXE generation failed then display an error and stop the CMAKE generation
+    if ( _workbench_powershell_return_code GREATER 0 )
+      message(${_workbench_powershell_output})
+      message(FATAL_ERROR "Generating the Workbench executable encountered an error.")
+    endif ()
+
+    # If the EXE generation succeeded then install the executable and rename it to remove the .install suffix
+    # the shortcuts for the workbench are ONLY installed if this variable has been declared
+    set (_workbench_executable_full_name ${_workbench_base_name}.exe)
+    message(STATUS "Generated the Workbench executable for installation: ${_workbench_executable_install_name}")
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${_workbench_executable_install_name} DESTINATION bin RENAME ${_workbench_executable_full_name})
+  else ()
+    message(STATUS "PowerShell was not found. Not generating the Workbench executable.")
```

Comment:
I'm tempted to say we should make this a `FATAL_ERROR` as if we can't generate the executable then the package is not that useful. What do you think?